libcontainer/cgroups/fs/hugetlb_test.go

```diff
@@ -32,38 +31,47 @@ func TestHugetlbSetHugetlb(t *testing.T) {
 		hugetlbAfter  = 512
 	)
 
-	helper.writeFileContents(map[string]string{
-		limit: strconv.Itoa(hugetlbBefore),
-	})
-
-	helper.CgroupData.c.HugetlbLimit = []*configs.HugepageLimit{
-		{
-			Pagesize: hugePageSize[0],
-			Limit:    hugetlbAfter,
-		},
-	}
-	hugetlb := &HugetlbGroup{}
-	if err := hugetlb.Set(helper.CgroupPath, helper.CgroupData.c); err != nil {
-		t.Fatal(err)
+	for _, pageSize := range HugePageSizes {
+		helper.writeFileContents(map[string]string{
+			fmt.Sprintf(limit, pageSize): strconv.Itoa(hugetlbBefore),
+		})
 	}
 
-	value, err := getCgroupParamUint(helper.CgroupPath, limit)
-	if err != nil {
-		t.Fatalf("Failed to parse %s - %s", limit, err)
+	for _, pageSize := range HugePageSizes {
+		helper.CgroupData.c.HugetlbLimit = []*configs.HugepageLimit{
+			{
+				Pagesize: pageSize,
+				Limit:    hugetlbAfter,
+			},
+		}
+		hugetlb := &HugetlbGroup{}
+		if err := hugetlb.Set(helper.CgroupPath, helper.CgroupData.c); err != nil {
+			t.Fatal(err)
+		}
 	}
-	if value != hugetlbAfter {
-		t.Fatalf("Set hugetlb.limit_in_bytes failed. Expected: %v, Got: %v", hugetlbAfter, value)
+
+	for _, pageSize := range HugePageSizes {
+		limit := fmt.Sprintf(limit, pageSize)
+		value, err := getCgroupParamUint(helper.CgroupPath, limit)
+		if err != nil {
+			t.Fatalf("Failed to parse %s - %s", limit, err)
+		}
+		if value != hugetlbAfter {
+			t.Fatalf("Set hugetlb.limit_in_bytes failed. Expected: %v, Got: %v", hugetlbAfter, value)
+		}
 	}
 }
 
 func TestHugetlbStats(t *testing.T) {
 	helper := NewCgroupTestUtil("hugetlb", t)
 	defer helper.cleanup()
-	helper.writeFileContents(map[string]string{
-		usage:    hugetlbUsageContents,
-		maxUsage: hugetlbMaxUsageContents,
-		failcnt:  hugetlbFailcnt,
-	})
+	for _, pageSize := range HugePageSizes {
+		helper.writeFileContents(map[string]string{
+			fmt.Sprintf(usage, pageSize):    hugetlbUsageContents,
+			fmt.Sprintf(maxUsage, pageSize): hugetlbMaxUsageContents,
+			fmt.Sprintf(failcnt, pageSize):  hugetlbFailcnt,
+		})
+	}
```

Comment:
I think for testing of runc api, test for one hugetlb page size is good enough, they are exactly the same for other page sizes.

I'm thinking about fix like this:
```
diff --git a/libcontainer/cgroups/fs/hugetlb_test.go b/libcontainer/cgroups/fs/hugetlb_test.go
index 39c3ee6..5b89a47 100644
--- a/libcontainer/cgroups/fs/hugetlb_test.go
+++ b/libcontainer/cgroups/fs/hugetlb_test.go
@@ -59,11 +59,18 @@ func TestHugetlbSetHugetlb(t *testing.T) {
 func TestHugetlbStats(t *testing.T) {
        helper := NewCgroupTestUtil("hugetlb", t)
        defer helper.cleanup()
-       helper.writeFileContents(map[string]string{
-               usage:    hugetlbUsageContents,
-               maxUsage: hugetlbMaxUsageContents,
-               failcnt:  hugetlbFailcnt,
-       })
+       // hugetlb.GetStats() will get stats for all HugePageSizes,
+       // we need to create them here.
+       for _, pageSize := range HugePageSizes {
+               usage := strings.Join([]string{"hugetlb", pageSize, "usage_in_bytes"}, ".")
+               maxUsage := strings.Join([]string{"hugetlb", pageSize, "max_usage_in_bytes"}, ".")
+               failcnt := strings.Join([]string{"hugetlb", pageSize, "failcnt"}, ".")
+               helper.writeFileContents(map[string]string{
+                       usage:    hugetlbUsageContents,
+                       maxUsage: hugetlbMaxUsageContents,
+                       failcnt:  hugetlbFailcnt,
+               })
+       }

        hugetlb := &HugetlbGroup{}
        actualStats := *cgroups.NewStats()
```

Anyway, test for all page sizes does no harm, it's even good for functional test, so I'm OK with your fix, thanks. :)